nf_core/pipelines/sync.py

```diff
@@ -273,17 +273,23 @@ def make_template_pipeline(self):
                 yaml.safe_dump(self.config_yml.model_dump(), config_path)
 
         try:
-            nf_core.pipelines.create.create.PipelineCreate(
+            pipeline_create_obj = nf_core.pipelines.create.create.PipelineCreate(
                 outdir=str(self.pipeline_dir),
                 from_config_file=True,
                 no_git=True,
                 force=True,
-            ).init_pipeline()
+            )
+            pipeline_create_obj.init_pipeline()
 
             # set force to false to avoid overwriting files in the future
             if self.config_yml.template is not None:
+                self.config_yml.template = pipeline_create_obj.config
                 # Set force true in config to overwrite existing files
                 self.config_yml.template.force = False
+                # Set outdir as the current directory to avoid local info leaking
+                self.config_yml.template.outdir = "."
```

Comment:
should we not set it back to the original value at the end? or do we actually not need that value after creation?

Reply:
We don't need it, and we always override it anyway. Here is the last step, where we write to the file, so I think it's safe to modify it here.